megamek/src/megamek/client/bot/princess/RankedPath.java

```diff
@@ -72,42 +74,43 @@ public int compareTo(RankedPath p) {
         if (p.rank < rank) {
             return 1;
         }
-        return Integer.compare(path.getHexesMoved(), p.path.getHexesMoved());
-    }
-
-    @Override
-    public boolean equals(Object o) {
-        if (this == o) {
-            return true;
-        }
-        if (o == null || getClass() != o.getClass()) {
-            return false;
+        if (path.getHexesMoved() < p.path.getHexesMoved()) {
+            return -1;
         }
-
-        RankedPath that = (RankedPath) o;
-
-        if (Double.compare(that.rank, rank) != 0) {
-            return false;
+        if (p.path.getHexesMoved() < path.getHexesMoved()) {
+            return 1;
         }
-        if (!path.equals(that.path)) {
-            return false;
+        if (expectedDamage < p.expectedDamage) {
+            return -1;
         }
-        if (reason != null ? !reason.equals(that.reason) : that.reason != null) {
-            return false;
+        if (p.expectedDamage < expectedDamage) {
+            return 1;

```

Comment:
Do we store the hazard values anywhere in the RankedPath?  
If so, we should probably compare those as well; if not, we might want to take a look at it for a later update.

Reply:
Thats a good idea, will take a look into it later